src/lib.rs

```diff
@@ -344,6 +344,8 @@ mod test {
                 .long_os_version()
                 .expect("Failed to get long OS version")
                 .is_empty());
+
+            assert!(!s.distribution_id().is_empty());
```

Comment:
Please move this test outside of the `if System::IS_SUPPORTED` since it can definitely work even on unknown targets.

Reply:
Done. Please take a look if this is what you meant.